### PR TITLE
DS-3575: Rename misguiding find method in ResourcePolicyService

### PR DESCRIPTION
**Pull request by pnbecker**
https://jira.duraspace.org/browse/DS-3575

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/authorize/ResourcePolicyServiceImpl.java [hunk 1]
```diff
@@ -104,13 +104,22 @@ public ResourcePolicy create(Context context) throws SQLException
     }
 
     @Override
-    public List<ResourcePolicy> find(Context c, DSpaceObject dso, Group group, int action, int notPolicyID) throws SQLException {
-        return resourcePolicyDAO.findByTypeIdGroupAction(c, dso, group, action, notPolicyID);
+    public List<ResourcePolicy> find(Context c, DSpaceObject dso, Group group, int action) throws SQLException {
+        return resourcePolicyDAO.findByTypeGroupAction(c, dso, group, action);
     }
     
+    // TODO: @Override
```
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/authorize/ResourcePolicyServiceImpl.java]: Not sure I understand the purpose to this TODO?  Did you mean to add an `@Override` here?

**Review by tdonohue — APPROVED** [review]
👍 Code looks good overall. I added a minor note about a "TODO" comment (to clarify its purpose), but beyond that I think this all looks good.  Obviously, this would need cherry-picking to all other applicable branches (minimally `master`).

File: dspace-api/src/main/java/org/dspace/authorize/ResourcePolicyServiceImpl.java (hunk 1 above)
Comment (pnbecker) [inline dspace-api/src/main/java/org/dspace/authorize/ResourcePolicyServiceImpl.java]: good catch, thanks.
Changed.

**Comment by pnbecker** [comment]
@tdonohue good catch. Thanks. I removed the comment and left the annotation.

**Review by hardyoyo — APPROVED** [review]
+1 by review

**Comment by pnbecker** [comment]
@tdonohue do you give me an official +1? I would like to see this getting merged.

**Comment by tdonohue** [comment]
@pnbecker : yes, a 👍 is a +1 ;)  Feel free to merge & cherry-pick to `master`. I meant to do this today, but didn't get to it. I'll do it tomorrow if you don't get to it first

**Pull request merged**